pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDReport.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+
+public class CPDReport {
+    private final Iterator<Match> matches;
+    private final Map<String, Integer> numberOfTokensPerFile;
+
+    CPDReport(final Iterator<Match> matches, final Map<String, Integer> numberOfTokensPerFile) {
+        this.matches = matches;
+        this.numberOfTokensPerFile = numberOfTokensPerFile;
+    }
+
+    public Iterator<Match> getMatches() {
```

Comment:
Since we are defining this new useful class: I find it rather unusual, to return an Iterator (I know, that's what CPD used to do)

Should we rather return a List here? In order to not change more files (CPD, MatchAlgorithm, ...), we could simply iterate through the matches in the constructor for now and create a new (unmodifiable) list in this class...

Reply:
I completely agree that a list would be preferable. My only reason against switching to a list would be that I would expect that to increase the memory requirement for CPD. I'm not sure if this is actually an issue though.

Reply:
I think memory-wise there will be no difference: The iterator needs to hold on internally to the original list anyway. The only difference is, that you can iterate only once - you have a one-shot report.

I'll try to make the actual `List<Match>` available here...